text-generation-inference/server/text_generation_server/jetstream_pt_support/engine_loader.py

```diff
@@ -19,12 +19,35 @@
 
 if TYPE_CHECKING:
     from transformers import PretrainedConfig
+
 from transformers import AutoConfig
 
 from .compatibility import model_can_use_jetstream_pt
 from .models import GemmaModel, LlamaModel, MixtralModel
 
 
+class OptimumJetstreamEngine(PyTorchEngine):
+    """This is essentially the same as the PytorchEngine, but it also supports a callback for sampling in prefill and
+    generation that can change on each request while not needing to be JIT'ed.
+    """
+    prefill_ex = PyTorchEngine.prefill
```

Comment:
Why renaming to prefill_ex?

Reply:
because I wanted to use the class original prefill code, but the PytorchEngine's `__init__` code replaces the object method with a JIT'ed version of the same: this causes a problem, because it requires the callback to be JIT'ed too, and I wanted to avoid using that. So the solution was to define another method, `prefill_ex`, that is essentially the same, but it is not jitted.